libraries/stumbler/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/motiondetection/MotionSensor.java

```diff
@@ -167,27 +181,28 @@ void start() {
             LocationManager lm = (LocationManager)mContext.getSystemService(Context.LOCATION_SERVICE);
             lm.requestLocationUpdates(LocationManager.GPS_PROVIDER, 0, 0, mListener);
 
-            handler.postDelayed(new Runnable() {
-                public void run() {
-                    stop();
-                }
-            }, 30 * 1000);
+            handler.postDelayed(mRunnable, 30 * 1000);
         }
 
         void stop() {
             LocationManager lm = (LocationManager)mContext.getSystemService(Context.LOCATION_SERVICE);
             lm.removeUpdates(mListener);
-            handler.removeCallbacks(null);
```

Comment:
that is legal. It removes all callbacks on the handler, but that was from WIP patch, I would avoid ever doing that in proper code